site/content/docs/howto/cicd.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Using Pinniped for CI/CD cluster operations
+description: Using Pinniped for CI/CD cluster operations.
+cascade:
+  layout: docs
+menu:
+  docs:
+    name: Use Pinniped for CI/CD
+    weight: 500
+    parent: howtos
+---
+
+This guide shows you how to configure Pinniped and your CI/CD system of choice to use Pinniped to administrate Kubernetes clusters.
+
+Note that the guide below assumes that you are using a non-human account within the IDP of your choice.
+It is never recommended to use a human's credentials in this way.
+
+## Prerequisites
+
+This how-to guide assumes that you have already configured the following Pinniped server-side components within your Kubernetes cluster(s):
+
+1. Pinniped Supervisor with a working FederationDomain and at least one IdentityProvider (LDAP, AD, or OIDC)
+   * The Supervisor installation could be on a completely separate cluster unrelated to your CI/CD
+2. Pinniped Concierge on each cluster that needs to be administrated by your CI/CD system
+   * It is possible to use the Pinniped CLI to log into any cluster configured with
+[OIDC authentication](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#openid-connect-tokens),
+see [here]({{< ref "../tutorials/supervisor-without-concierge-demo" >}}). This would not require Concierge to be installed
+on each cluster.
+3. A CI/CD system that meets the following conditions:
+   * It can handle secrets safely and provide them to tasks as environment variables
+   * It can run shell scripts, or at least invoke binaries (such as `pinniped` and `kubectl`) 
+   * It can access Pinniped-style kubeconfigs for each cluster that needs to be administrated
+4. A user account (that does not represent a human) within the IDP of your choice
+   * This account should be granted the least amount of privileges necessary
+   * This account should likely be single-purposed for CI/CD administration
+
+## Overview
+
+1. A CI/CD admin should generate the Pinniped-style kubeconfig for each cluster that needs to be administered by CI/CD and make those kubeconfigs available to CI/CD 
```

Comment:
Should this discuss the difference between browser flow and CLI flow? They need to choose CLI flow when creating the kubeconfig, or they need to override the flow specified in the kubeconfig by using an env var at authentication time.

Reply:
Good point. I had thought that `PINNIPED_USERNAME/PASSWORD` would force the `cli_password` flow but it's best to call that out explicitly.